### PR TITLE
feat: add pre-commit

### PR DESCRIPTION
**Pull request by aarnphm**
Signed-off-by: Aaron Pham <29749331+aarnphm@users.noreply.github.com>

## Checklist:

- [ ] My code follows the bentoml code style, both `./dev/format.sh` and
  `./dev/lint.sh` script have passed
  ([instructions](https://github.com/bentoml/BentoML/blob/master/DEVELOPMENT.md#style-check-and-auto-formatting-your-code)).
- [ ] My change reduces project test coverage and requires unit tests to be added
- [ ] I have added unit tests covering my code change
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
```
Comment (parano) [inline README.md:2]: let's remove changes in docs?

Comment (aarnphm) [reply README.md:2]: oh i just enable `end-of-file-fixer` under pre-commit. Should we disable that?

Comment (parano) [reply README.md]: Probably, doesn't it mess up the format in rst or md files?

Comment (aarnphm) [reply README.md]: i don't think so, but we can chat more

**Comment by sauyon** [comment]
Do we need to use pre-commit and not just have a provided pre-commit hook?

**Comment by aarnphm** [comment]
> Do we need to use pre-commit and not just have a provided pre-commit hook?

I think it is a lot nicer to use pre-commit, since it is easier to maintain imo

**Comment by aarnphm** [comment]
closed this since it is obsolete now

**Pull request closed without merging**